lib/src/main/java/io/ably/lib/transport/ConnectionManager.java

```diff
@@ -269,7 +269,7 @@ public void run() {
 	 * transport events/notifications
 	 ***************************************/
 
-	void onMessage(ProtocolMessage message) throws AblyException {
+	public void onMessage(ProtocolMessage message) throws AblyException {
```

Comment:
> It's a shame we need to make this public just in order to be able to call it from a test. I can't think of an alternative at the moment without mocking the transport so the message comes from there.

I don't know if there is a convention if Java, but in Ruby you can add `@api private` to the method so that at least it's documented clearly that the method is public, but not as part of the public API